sdk/digitaltwins/Azure.DigitalTwins.Core/src/Models/QueryResult.cs

```diff
@@ -54,6 +52,21 @@ internal static QueryResult<T> DeserializeQueryResult(JsonElement element, Objec
         {
             IReadOnlyList<T> items = default;
             string continuationToken = default;
+
+            // If the provided objectSerializer is of type JsonObjectSerializer, we will re-use the same object and set it as the defaultObjectSerializer.
+            if (objectSerializer is JsonObjectSerializer)
+            {
+                defaultObjectSerializer = objectSerializer;
+            }
+            // Otherwise, if the defaultObjectSerializer is null, we will statically instantiate it and re-use it in the future.
+            else
```

Comment:
this can be an else if

Reply:
I moved the logic to DigitalTwinsClient
but not sure why you think it should be else if, take a look at it in the new file and we can discuss further.

Reply:
No longer relevant.